model_hub/examples/huggingface/translation/translation_trial.py

```diff
@@ -0,0 +1,348 @@
+import model_hub.utils as utils
+
+# A list of all multilingual tokenizer which require src_lang and tgt_lang attributes.
+MULTILINGUAL_TOKENIZERS = [
+    transformers.MBartTokenizer,
+    transformers.MBartTokenizerFast,
+    transformers.MBart50Tokenizer,
+    transformers.MBart50TokenizerFast,
+    transformers.M2M100Tokenizer,
+]
+
+
+class TranslationTrial(hf.BaseTransformerTrial):
+    def __init__(self, context: det_torch.PyTorchTrialContext) -> None:
+        self.logger = logging.getLogger(__name__)
+        self.hparams = attrdict.AttrDict(context.get_hparams())
+        self.data_config = attrdict.AttrDict(context.get_data_config())
+
+        if (
+            self.data_config.source_prefix is None
+            and self.hparams.pretrained_model_name_or_path
+            in [
+                "t5-small",
+                "t5-base",
+                "t5-large",
+                "t5-3b",
+                "t5-11b",
+            ]
+        ):
+            self.logger.warning(
+                "You're running a t5 model but didn't provide a source prefix, which is expected, "
+                "e.g. with `data.source_prefix 'translate English to German: ' `"
+            )
+        # Creates HF config, tokenizer, model using parent init function.
+        super(TranslationTrial, self).__init__(context)
+        self.logger.info(self.config)
+
+        # Prep dataset
+        # Get the datasets: you can either provide your own CSV or JSON training and evaluation
+        # files (see below) or just provide the name of one of the public datasets available on the
+        # hub at https://huggingface.co/datasets/ (the dataset will be downloaded automatically
+        # from the datasets Hub).
+
+        # For CSV/JSON files, this script will use the column called 'text' or the first column if
+        # no column called 'text' is found. You can easily tweak this behavior (see below).
+
+        # See more about loading any type of standard or custom dataset (from files, python dict,
+        # pandas DataFrame, etc) at
+        # https://huggingface.co/docs/datasets/loading_datasets.html.
+        if self.data_config.val_max_target_length is None:
+            self.data_config.val_max_target_length = min(
+                self.data_config.max_target_length, self.tokenizer.model_max_length
+            )
+        self.raw_datasets = hf.default_load_dataset(self.data_config)
+        self.tokenized_datasets = self.build_datasets()
+
+        train_length = len(self.tokenized_datasets["train"])
+        self.logger.info("training records: {}".format(train_length))
+        if (
+            "records_per_epoch" in self.exp_config
+            and train_length != self.exp_config["records_per_epoch"]
+        ):
+            self.logger.warning(
+                "number of train records {} does not match records_per_epoch of {}".format(
+                    train_length, self.exp_config["records_per_epoch"]
+                )
+            )
```

Comment:
I think this block here is assuming that you are only running for 1 epoch which is true for all of the default configs but not necessarily for all user-provided configs, so should it be:
`train_length != self.exp_config["records_per_epoch"] * self.exp_config["epochs"]`

Reply:
i don't think we should be multiplying `records_per_epoch` by the number of epochs.  `train_length` is the length of the dataset so it should be the same as `records_per_epoch` if that field is set in the experiment config.